src/main/java/de/caritas/cob/agencyservice/api/admin/service/agency/AgencyAdminSearchTenantSupportService.java

```diff
@@ -0,0 +1,59 @@
+package de.caritas.cob.agencyservice.api.admin.service.agency;
+
+import static de.caritas.cob.agencyservice.api.repository.agency.Agency.SEARCH_ANALYZER;
+
+import de.caritas.cob.agencyservice.api.repository.agency.Agency;
+import de.caritas.cob.agencyservice.api.tenant.TenantContext;
+import javax.persistence.EntityManagerFactory;
+import lombok.NonNull;
+import org.apache.lucene.queryparser.classic.QueryParserBase;
+import org.apache.lucene.search.Query;
+import org.hibernate.search.jpa.FullTextEntityManager;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Service;
+
+@Service
+@Primary
+@ConditionalOnExpression("${multitenancy.enabled:true}")
+public class AgencyAdminSearchTenantSupportService extends AgencyAdminSearchService {
+
+  public AgencyAdminSearchTenantSupportService(
+      @NonNull EntityManagerFactory entityManagerFactory) {
+    super(entityManagerFactory);
+  }
+
+  protected Query buildUnfilteredQuery(FullTextEntityManager fullTextEntityManager) {
+    return fullTextEntityManager.getSearchFactory()
+        .buildQueryBuilder()
+        .forEntity(Agency.class)
+        .get().keyword().onField(TENANT_ID_SEARCH_FIELD)
+        .matching(TenantContext.getCurrentTenant()).createQuery();
+  }
+
+  protected Query buildFullTextSearchQuery(String keyword, FullTextEntityManager entityManager) {
+    var queryBuilder = entityManager.getSearchFactory()
+        .buildQueryBuilder()
+        .forEntity(Agency.class)
+        .get();
+
+    return entityManager.getSearchFactory()
```

Comment:
Please remove all not-tenant relevant logic from this class (so base class provides all the search business logic and tenant class enriches the builder with only tenant specific parts)

Reply:
I am not sure that we should split this query across two classes. 

I understand your point that maybe in the future if someone changes a part in one query, then it could happen that he forgets the tenant query, but then this same logic would apply for native queries, where we also have two versions of the query. We would then also need to make this queries more generic.

Since we have to merge this changes back to the public repository, this approach is much cleaner and easier to integrate.  
We don't change the structure of the existing service at all. We just added a new class. If we implement your suggestion then we would need to change the signature of the methods, hence instead of queries return querybuilder. This code would not be so clear and readable like the one that we have, and we would not have a clean separation from tenant feature from the core of the application.


Do you agree?

Reply:
okay, taken into account effort to retrofit to open source it's better to keep it like this for now